### ISSUE TITLE
Namespace local configuration does not silence :missing-else-branch

### ISSUE THREAD

**Issue by khardenstine** [clj-kondo/clj-kondo#2181]
**version**

clj-kondo v2023.09.07

**problem**

:missing-else-branch cannot be ignored via namespace local configuration

**repro**

```clj
(ns repro
  {:clj-kondo/config {:ignore true}})

(if 1 2)

(not (empty? nil))

#_{:clj-kondo/ignore true}
(if 1 2)

#_{:clj-kondo/ignore [:missing-else-branch]}
(if 1 2)
```

```sh
clj-kondo --lint repro.clj
repro.clj:4:1: warning: Missing else branch.
linting took 8ms, errors: 0, warnings: 1
```

Also reproduced with ns metadata:
```clj
{:clj-kondo/config {:linters {:missing-else-branch {:level :off}
                              :not-empty?          {:level :off}}}}
```

**config**

Reproduced as well with these configs instead of ns metadata:

```clj
{:config-in-ns {repro {:ignore true}}}
```

```clj
{:config-in-ns {repro {:ignore [:missing-else-branch
                                :not-empty?]}}}
```

**expected behavior**

no warning
